nri-redis/src/Redis/Mock.hs

```diff
@@ -47,21 +49,18 @@ handlerIO = do
     |> Prelude.pure
   where
     doQuery' modelRef doAnything = \query ->
-      atomicModifyIORef'
-        modelRef
-        ( \model ->
-            let (newHash, res) = doQuery query (hash model)
-             in ( model {hash = newHash},
-                  res
-                )
-        )
+      atomicModifyIORef' modelRef (doQuery query)
         |> Platform.doAnything doAnything
         |> Internal.traceQuery (Internal.cmds query) "Redis.Mock" Nothing
 
 -- | This is our mock implementation of the Redis state. Our mock implementation
 -- will store a single value of this type, and redis commands will modify it.
 data Model = Model
-  {hash :: HM.HashMap Text RedisType}
+  { hash :: HM.HashMap Text RedisType,
+    scans :: Scans
```

Comment:
We need some internal state to implement a `scan` that works correctly when used to delete keys during the scan! Using the cursor as the only form of state doesn't work (or at least I couldn't see a way)